lib/script/gql_generate.ts

```diff
@@ -28,14 +28,35 @@ export async function generateGql(options: {
 
 	// Load globs from the codegen.yaml
 	const yaml = await import("js-yaml");
+	const localCodegenPath = path.join(options.cwd, "codegen.yml");
+	const graphqlCodegenPath = path.join(options.cwd, "graphql", "codegen.yml");
+	const npmCodegenPath = path.join(
+		options.cwd,
+		"node_modules",
+		"@atomist",
+		"skill",
+		"graphql",
+		"codegen.yaml",
+	);
+	const skillCodegenPath = path.join(
```

Comment:
Why would we ever want to use the config from the root of the skill project? Is that so that we can use this on itself?

Reply:
None of these point to the codegen.yml in the root of the skill project. Both `npmCodegenPath` and `skillCodegenPath` point to the `graphql/codegen.yaml` in an `@atomist/skill` package. In most cases, the `npmCodegenPath` and `skillCodegenPath` will resolve to the same `@atomist/skill/graphql/codegen.yaml`. If, however, the user is using a globally installed `atm-skill` or one installed in another project by providing the `cwd` option, having `npmCodgenPath` ensures the config provided by the project's `@atomist/skill` version be used as opposed to the one provided by the version of `@atomist/skill` installed elsewhere.

Previously, the equivalent of `skillCodegenPath` was used to determine if any files matched the globs (https://github.com/atomist-skills/skill/pull/143/files#diff-2b129237ae2f9def726f9fc4cabfc97c6eb3b8cd2c8f082f988829a0f5db5d38L33) and the equivalent of `npmCodegenPath` was used when running the `@graphql-codegen/cli` (https://github.com/atomist-skills/skill/pull/143/files#diff-2b129237ae2f9def726f9fc4cabfc97c6eb3b8cd2c8f082f988829a0f5db5d38L54-L61). Since they both appeared previously, I preserved them both.

Reply:
Right, I misread that. Sorry. Approving now.